include/dgl/runtime/ndarray.h

```diff
@@ -147,39 +147,42 @@ class NDArray {
     else
       return static_cast<T*>(operator->()->data);
   }
-  /*!
-   * \brief Copy data content from another array.
+   /*!
+   * \brief Copy data content from/into another array.
    * \param other The source array to be copied from.
-   * \param stream The stream to perform the copy on if it involves a GPU
-   *        context, otherwise this parameter is ignored.
-   * \note The copy may happen asynchrously if it involves a GPU context.
-   *       DGLSynchronize is necessary.
+   * \note The copy runs on the dgl internal stream if it involves a GPU context.
    */
-  inline void CopyFrom(DLTensor* other,
-      DGLStreamHandle stream = nullptr);
-  inline void CopyFrom(const NDArray& other,
-      DGLStreamHandle stream = nullptr);
+  inline void CopyFrom(DLTensor* other);
+  inline void CopyFrom(const NDArray& other);
+  inline void CopyTo(DLTensor *other) const;
+  inline void CopyTo(const NDArray &other) const;
+
   /*!
-   * \brief Copy data content into another array.
+   * \brief Asynchronous copy data content from/into another array
+   *        with a specified stream (GPU only).
    * \param other The source array to be copied from.
-   * \note The copy may happen asynchrously if it involves a GPU context.
-   *       DGLSynchronize is necessary.
-   */
-  inline void CopyTo(DLTensor *other,
-                     const DGLStreamHandle &stream = nullptr) const;
-  inline void CopyTo(const NDArray &other,
-                     const DGLStreamHandle &stream = nullptr) const;
+   * \param stream The stream to perform the copy context,
+   * \note The copy may happen synchronously if specified stream = dgl internal stream.
+   *       Otherwise, synchronization is necessary.
+   */
+  inline void CopyFromAsync(DLTensor* other,
```

Comment:
I doubt if we really need `CopyFromAsync`/`CopyToAsync`. Where will they be used?

Reply:
Not really. This is mostly for future I guess. Maybe I can comment/mark it as `Todo`...